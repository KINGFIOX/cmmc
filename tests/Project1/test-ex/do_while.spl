int main() {
    int i = 0;
    do {
        write(i);
        i = i + 1;
    } while(i < 10);
    return i;
}
