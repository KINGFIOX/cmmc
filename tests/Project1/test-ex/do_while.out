Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: main
        LP
        RP
      CompSt (1)
        LC
        DefList (2)
          Def (2)
            Specifier (2)
              TYPE: int
            DecList (2)
              Dec (2)
                VarDec (2)
                  ID: i
                ASSIGN
                Exp (2)
                  INT: 0
            SEMI
        StmtList (3)
          Stmt (3)
            DO
            Stmt (3)
              CompSt (3)
                LC
                StmtList (4)
                  Stmt (4)
                    Exp (4)
                      Exp (4)
                        ID: write
                      LP
                      Args (4)
                        Exp (4)
                          ID: i
                      RP
                    SEMI
                  StmtList (5)
                    Stmt (5)
                      Exp (5)
                        Exp (5)
                          ID: i
                        ASSIGN
                        Exp (5)
                          Exp (5)
                            ID: i
                          PLUS
                          Exp (5)
                            INT: 1
                      SEMI
                RC
            WHILE
            LP
            Exp (6)
              Exp (6)
                ID: i
              LT
              Exp (6)
                INT: 10
            RP
            SEMI
          StmtList (7)
            Stmt (7)
              RETURN
              Exp (7)
                ID: i
              SEMI
        RC
